core/lib/src/rocket.rs

```diff
@@ -33,6 +33,18 @@ use crate::http::uri::Origin;
 /// The main `Rocket` type: used to mount routes and catchers and launch the
 /// application.
 pub struct Rocket {
+    pub(crate) inner: Option<RocketInner>,
+    pending: Vec<BuildOperation>,
+}
+
+enum BuildOperation {
+    Mount(Origin<'static>, Vec<Route>),
+    Register(Vec<Catcher>),
+    Manage(Box<dyn FnOnce(RocketInner) -> RocketInner + Send + Sync + 'static>),
+    Attach(Box<dyn Fairing>),
+}
+
+pub(crate) struct RocketInner {
```

Comment:
The name `RocketInner`, for whatever reason, doesn't sit well with me. Could we name it something more indicative of the fact that it contains all of the information about the application? `Manifest`, perhaps, in keeping with the theme? Then we can have `rocket.manifest().await`.

Reply:
I like the type name `Manifest`. I'm not quite sold on the method name `manifest()` because of potential confusion over "to *manifest* a `Rocket`" vs "to retrieve a `Rocket`'s *manifest*`.